app/components/secure/WalletSecureComponent.tsx

```diff
@@ -2,88 +2,64 @@ import * as React from 'react';
-import { FragmentMediaContent } from '../../components/FragmentMediaContent';
-import { useReboot } from '../../utils/RebootContext';
+import { RoundButton } from '../RoundButton';
+import { FragmentMediaContent } from '../FragmentMediaContent';
 import { t } from '../../i18n/t';
-import { systemFragment } from '../../systemFragment';
 import { warn } from '../../utils/log';
-import { deriveUtilityKey } from '../../storage/utilityKeys';
 import { useAppConfig } from '../../utils/AppConfigContext';
 import { useTypedNavigation } from '../../utils/useTypedNavigation';
+import { mnemonicToWalletKey } from 'ton-crypto';
+import { contractFromPublicKey } from '../../engine/contractFromPublicKey';
+import { getAppState, getBackup, markAddressSecured } from '../../storage/appState';
+import { useReboot } from '../../utils/RebootContext';
 
-export const WalletSecureFragment = systemFragment((props: { mnemonics: string, deviceEncryption: DeviceEncryption, import: boolean }) => {
-    const { Theme, AppConfig } = useAppConfig();
+export const WalletSecureComponent = React.memo((props: {
+    mnemonics: string,
+    deviceEncryption: DeviceEncryption,
+    callback: (res: boolean) => void,
+    onLater?: () => void,
+    import?: boolean
+}) => {
+    const reboot = useReboot();
     const navigation = useTypedNavigation();
+    const { Theme, AppConfig } = useAppConfig();
     const safeArea = useSafeAreaInsets();
-    const reboot = useReboot();
+    let disableEncryption =
+        props.deviceEncryption === 'none'
+        || props.deviceEncryption === 'device-biometrics'
+        || props.deviceEncryption === 'device-passcode'
+        || Platform.OS === 'android' && Platform.Version < 30 // Encryption doesn't work well on older androids
 
     // Action
     const [loading, setLoading] = React.useState(false);
-    const onClick = React.useCallback((bypassEncryption?: boolean) => {
+    const onClick = React.useCallback(() => {
         (async () => {
             setLoading(true);
             try {
 
-                // Encrypted token
-                let secretKeyEnc: Buffer;
-
                 // Generate New Key
-                try {
-                    let disableEncryption = !!(props.deviceEncryption === 'none' || props.deviceEncryption === 'device-biometrics' || props.deviceEncryption === 'device-passcode' || bypassEncryption);
-                    if (Platform.OS === 'android' && Platform.Version < 30) {
-                        disableEncryption = true; // Encryption doesn't work well on older androids
-                    }
-                    secretKeyEnc = await generateNewKeyAndEncrypt(disableEncryption, Buffer.from(props.mnemonics));
-                } catch (e) {
-                    // Ignore
-                    console.warn(e);
-                    return;
-                }
+                let secretKeyEnc = await generateNewKeyAndEncrypt(
+                    false,
```

Comment:
are you sure that it won't fail if encryption should be disabled?

Reply:
onClick will not be invoked, because we leave this screen when disableEncryption is true and we are after import, also we will not navigate to this screen on init if encryption should be disabled.